File: backspaceCompare.py
class Solution:
    def processStr(self, str: str) -> str:
        newArr = []

        for char in str.split(""):
            if char == '#':
                newArr.pop()
            else:
                newArr.append(char)

        return "".join(newArr)

    def backspaceCompare(self, S: str, T: str) -> bool:
        return processStr(S) == processStr(T)

File: backspace_compare.rb
def process_str(str)
  new_arr = []
  str.each_char do |char|
    if char == '#'
      new_arr.pop
    else
      new_arr << char
    end
  end
  new_arr.join
end

def backspace_compare(s, t)
  process_str(s) == process_str(t)
end
